xrootd/cmd/xrd-srv/handler.go

```diff
@@ -181,36 +202,154 @@ func (h *handler) Close(sessionID [16]byte, request *xrdclose.Request) (xrdproto
 
 // Read implements server.Handler.Read.
 func (h *handler) Read(sessionID [16]byte, request *read.Request) (xrdproto.Marshaler, xrdproto.ResponseStatus) {
-	h.mu.RLock()
-	sess, ok := h.sessions[sessionID]
-	h.mu.RUnlock()
-	if !ok {
-		// This situation can appear if user tries to read without opening any file at all.
+	file := h.getFile(sessionID, request.Handle)
+	if file == nil {
+		return xrdproto.ServerError{
+			Code:    xrdproto.InvalidRequest,
+			Message: fmt.Sprintf("Invalid file handle: %v", request.Handle),
+		}, xrdproto.Error
+	}
+
+	buf := make([]byte, request.Length)
+	_, err := file.ReadAt(buf, request.Offset)
```

Comment:
From the documentation for `File.ReadAt`:
```
// ReadAt always returns a non-nil error when n < len(b).
```
So, I think that it's not really necessary since in that case `io.ReadFull` will produce the same result (`n` readed bytes and an error).

Reply:
Ok.